### PR TITLE
Update to .NET 5 preview 8

### PR DESCRIPTION
**Pull request by rjmholt**
# PR Summary

Updates our .NET SDK dependency to 5-preview.8.

Also fixes issues with csproj files for powershell-win-core and GraphicalHost that blocked this update.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/powershell-win-core/powershell-win-core.csproj [hunk 1]
```diff
@@ -64,4 +57,6 @@
     <ProjectReference Include="..\Microsoft.Management.UI.Internal\Microsoft.PowerShell.GraphicalHost.csproj" Condition="'$(SDKToUse)' == 'Microsoft.NET.Sdk.WindowsDesktop' "/>
   </ItemGroup>
 
+  <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
```
Comment (daxian-dbw) [inline src/powershell-win-core/powershell-win-core.csproj:60]: Shall we move this above next to `<Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />`?

File: src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
   <Import Project="..\..\PowerShell.Common.props" />
   <PropertyGroup>
     <Description>Assembly containing WPF code for Out-GridView, HelpWindow, and Show-Command</Description>
     <NoWarn>$(NoWarn);CS1570</NoWarn>
     <RootNamespace>Microsoft.Management.UI.Internal</RootNamespace>
     <AssemblyName>Microsoft.PowerShell.GraphicalHost</AssemblyName>
     <UseWPF>True</UseWPF>
+    <TargetPlatformIdentifier>Windows</TargetPlatformIdentifier>
+    <TargetPlatformVersion>7.0</TargetPlatformVersion>
```
Comment (daxian-dbw) [inline src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj:10]: Just curious, what does the `TargetPlatformVersion` mean? Meaning targeting win-7?

Comment (SteveL-MSFT) [reply src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj:10]: Yes, that is the OS version

Comment (SteveL-MSFT) [reply src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj:10]: We might have to have separate Win7 and Win10 packages as I think it's required to set this to 10 to pull in the WinRT packages

File: DotnetRuntimeMetadata.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
   "sdk": {
```
Comment (SteveL-MSFT) [inline DotnetRuntimeMetadata.json:2]: Doesn't global.json need to be updated?  Building this branch locally still uses preview.7

File: src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj (hunk 2 above)
Comment (daxian-dbw) [reply src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj:10]: Can we try setting this to 10 in a private branch and see if that helps the WinRT issue?

File: DotnetRuntimeMetadata.json (hunk 3 above)
Comment (rjmholt) [reply DotnetRuntimeMetadata.json:2]: That's now been updated, by running `.\tools\UpdateDotnetRuntime.ps1`

File: src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj (hunk 2 above)
Comment (SteveL-MSFT) [reply src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj:10]: I'm doing a local build to verify this

**Comment by rjmholt** [comment]
Waiting on the wix file to generate

File: src/powershell-win-core/powershell-win-core.csproj (hunk 1 above)
Comment (adityapatwardhan) [reply src/powershell-win-core/powershell-win-core.csproj:60]: We can try, but this was the recommendation from the .NET team member.

File: src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj (hunk 2 above)
Comment (adityapatwardhan) [reply src/Microsoft.Management.UI.Internal/Microsoft.PowerShell.GraphicalHost.csproj:10]: I think it is too late in the release to generate 2 packages for windows. We should do that for 7.2 if needed.

File: PowerShell.Common.props [hunk 4]
```diff
@@ -110,6 +110,9 @@
     <AssemblyOriginatorKeyFile>../signing/visualstudiopublic.snk</AssemblyOriginatorKeyFile>
     <SignAssembly>true</SignAssembly>
     <HighEntropyVA>true</HighEntropyVA>
+
+    <!--<NoWarn>$(NoWarn);SYSLIB0001</NoWarn>-->
+    <EnableUnsafeUTF7Encoding>true</EnableUnsafeUTF7Encoding>
```
Comment (adityapatwardhan) [inline PowerShell.Common.props:115]: Remove this.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Comment by rjmholt** [comment]
Back to draft, since builds from this branch seem unable to start on Windows due to an exception here:

https://github.com/PowerShell/PowerShell/blob/4e2b6e9540c1946d22664149f0331447ac0bf797/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs#L586-L605

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 5]
```diff
@@ -7617,9 +7617,7 @@ public Encoding Encoding
 
             set
             {
-#pragma warning disable MSLIB0001
-                if (value == System.Text.Encoding.UTF7)
-#pragma warning restore MSLIB0001
+                if (value != null && value.CodePage == 65000)
```
Comment (daxian-dbw) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs:7620]: I guess `65000` code page stands for UTF7, right? Can you please add a comment?

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs [hunk 6]
```diff
@@ -2576,7 +2576,9 @@ internal static void WriteConsole(ConsoleHandle consoleHandle, ReadOnlySpan<char
                     {
                         var endOfLine = Environment.NewLine.AsSpan();
                         var endOfLineLength = endOfLine.Length;
+#pragma warning disable CA2014
                         Span<char> outBufferLine = stackalloc char[outBuffer.Length + endOfLineLength];
+#pragma warning restore CA2014
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs:2581]: Shall we open issues tracking this? It looks we shouldn't use `stsackalloc` in a real loop (`do ... while(false)` is fine).

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs (hunk 5 above)
Comment (rjmholt) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs:7620]: Will do. Also see https://github.com/dotnet/runtime/issues/41459#issuecomment-682106551

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs (hunk 6 above)
Comment (rjmholt) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs:2581]: See https://github.com/PowerShell/PowerShell/issues/13531

**Comment by iSazonov** [comment]
@DHowett Have you any thoughts about https://github.com/PowerShell/PowerShell/pull/13530#issuecomment-682130829 ?

File: src/System.Management.Automation/utils/EncodingUtils.cs [hunk 7]
```diff
@@ -8,6 +8,7 @@
 
 using System.Management.Automation.Internal;
 
+
```
Comment (daxian-dbw) [inline src/System.Management.Automation/utils/EncodingUtils.cs:11]:
```suggestion
```

Comment (daxian-dbw) [reply src/System.Management.Automation/utils/EncodingUtils.cs:11]: This is to fix the CodeFactor issue.

**Review by adityapatwardhan — APPROVED** [review]

**Review by iSazonov — APPROVED** [review]

**Pull request merged**